src/hooks/use-aggrid-sort-impl.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { AgGridReact } from 'ag-grid-react';
+import { useCallback, useEffect } from 'react';
+import {
+    SortConfigType,
+    SortPropsType,
+    getSortValue,
+    useAgGridSort,
+} from './use-aggrid-sort';
+
+export const useAgGridSortImpl = (
+    gridRef: React.MutableRefObject<AgGridReact | null>,
+    initSortConfig: SortConfigType
+): SortPropsType => {
+    const { onSortChanged, sortConfig, initSort } =
+        useAgGridSort(initSortConfig);
+
+    const setSortInAgGrid = useCallback(
+        (sortConfig: SortConfigType) => {
+            gridRef.current?.columnApi?.applyColumnState({
+                state: [
+                    {
+                        colId: sortConfig.colKey,
+                        sort: getSortValue(sortConfig.sortWay),
+                    },
+                ],
+                defaultState: { sort: null },
+            });
+        },
+        [gridRef]
```

Comment:
it's a bit weird to pass a ref as useCallback dependency but I did not find any way to do without it

Reply:
Yes, just to avoid eslint error